src/libvirt-xml-update.ts

```diff
@@ -286,27 +339,28 @@ export function updateBootOrder(domXml, devices) {
 /*
  * This function is used to define only offline attribute of memory.
  */
-export function updateMaxMemory(domXml, maxMemory) {
+export function updateMaxMemory(domXml: string, maxMemory: number): string {
     const doc = getDoc(domXml);
     const domainElem = doc.firstElementChild;
     const s = new XMLSerializer();
 
-    const memElem = domainElem.getElementsByTagName("memory")[0];
-    memElem.textContent = `${maxMemory}`;
+    const memElem = domainElem?.getElementsByTagName("memory")[0];
+    memElem!.textContent = `${maxMemory}`;
```

Comment:
This reads rather funny :) We should imo assert this instead? so I'd say cockpit.assert it also raises a exception with a message.

Reply:
This should probably add the right element when `memElem` is null...

Reply:
Ah, let's just put an assert here.